### PR TITLE
Wireguard

### PR DESCRIPTION
**Pull request by AlexanderMalmstrom**
Allt kommer att bli bra

### CODE REVIEW

File: modules/wireguard/manifests/init.pp [hunk 1]
```diff
@@ -1,68 +1,96 @@
...
+    unless  => '/usr/bin/ls /etc/wireguard/privkey',
+    require => Exec['enable-forward'],
   }
 
+  # Create wireguard pubkey
   exec { 'create-pubkey':
-    command => '/usr/bin/wg genkey > /etc/wireguard/privkey',
-    unless  => '/usr/bin/ls /etc/wireguard/privkey'
+    command => '/usr/bin/wg pubkey < /etc/wireguard/privkey > /etc/wireguard/pubkey',
+    unless  => '/usr/bin/ls /etc/wireguard/pubkey',
```
Comment (SoundGoof) [inline modules/wireguard/manifests/init.pp]: exec { 'create-pubkey':
  command => '/usr/bin/wg pubkey < /etc/wireguard/privkey > /etc/wireguard/pubkey',
  creates => '/etc/wireguard/pubkey',
  require => Exec['create-privkey'],

File: modules/wireguard/metadata.json [hunk 2]
```diff
@@ -0,0 +1,13 @@
+{
+    "name": "dhtech-WireGuard ",
```
Comment (SoundGoof) [inline modules/wireguard/metadata.json]: why extra space?

File: modules/wireguard.py [hunk 3]
```diff
@@ -1,16 +1,52 @@
...
+    if not res:
+        raise NodeNotFoundError('Node %s not found' % host)
+
+    netmask = res[0]
+
+    db.execute('SELECT ipv4_gateway_txt FROM network WHERE short_name = "TECH-WIREGUARD-VPN";')
+    res = db.fetchone()
+    conn.close()
+    if not res:
+        raise NodeNotFoundError('Node %s not found' % host)
```
Comment (SoundGoof) [inline modules/wireguard.py]: you cant raise here, it will only be error 500 from puppet

Comment (SoundGoof) [reply modules/wireguard.py]: do empty and check in puppet module if you get data in variable instead

File: modules/wireguard.py [hunk 4]
```diff
@@ -1,16 +1,44 @@
...
+def generate(host, *args): 
+    # Get current event
     current_event = lib.get_current_event()
 
+    if os.path.isfile(DB_FILE):
+        try:
+            conn = sqlite3.connect(DB_FILE)
+            db = conn.cursor()
+        except sqlite3.Error as e:
+            print("An error occurred:", e.args[0])
```
Comment (SoundGoof) [inline modules/wireguard.py]: silent errors here

File: modules/wireguard.py [hunk 5]
```diff
@@ -1,16 +1,49 @@
 # Copyright 2024 dhtech
-#
 # Use of this source code is governed by a BSD-style
 # license that can be found in the LICENSE file
+
 import lib
+import sqlite3
+import os
+import ipcalc
+import sys
```
Comment (SoundGoof) [inline modules/wireguard.py]: is sys used?

Comment (AlexanderMalmstrom) [reply modules/wireguard.py]: Sys is indeed not used anymore, I've removed it.

**Review by SoundGoof — APPROVED** [review]
lgtm

**Pull request merged**